include/velocypack/Slice.h

```diff
@@ -117,36 +122,108 @@ class Slice {
   // creates a slice of type MaxKey
   static constexpr Slice maxKeySlice() noexcept { return Slice(maxKeySliceData); }
 
-  // pointer to the head byte
-  constexpr uint8_t const* start() const noexcept { return _start; }
-  
+  constexpr Slice const value() const noexcept {
+    return isTagged() ? Slice(valueStart()) : *this;
+  }
+
+  constexpr uint64_t getFirstTag() const {
+    // always need the actual first byte, so use _start directly
+    return !isTagged() ? 0 :
+             (*_start == 0xee ? readIntegerFixed<uint64_t, 1>(_start + 1) :
+               (*_start == 0xef ? readIntegerFixed<uint64_t, 8>(_start + 1) : /* error */ 0)
+             );
+  }
+
+  std::list<uint64_t> getTags() const {
```

Comment:
I think an `std::vector<uint64_t>` would be more lightweight here.
`std::list` needs to allocate each member seperately, whereas a vector doesn't need allocations per element (effectively it will do a few allocations for the first few elements, but afterwards it will reallocate a lot less).
Apart from that, using vector here allows removing the `<list>` include from the header and reduces dependencies.